src/main/java/net/minecraftforge/common/capabilities/ICapabilityFactory.java

```diff
@@ -0,0 +1,7 @@
+package net.minecraftforge.common.capabilities;
+
+import net.minecraft.resources.ResourceLocation;
+
+public interface ICapabilityFactory<T> {
```

Comment:
You should mark this interface with `@FunctionalInterface` since these factories are effectively lambdas that can be used with the system individually.